src/validation/decorators/is_node_notify_object.ts

```diff
@@ -0,0 +1,86 @@
+/*
+* Copyright 2021 HMS Industrial Networks AB
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http: //www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+import {
+    registerDecorator,
+    ValidationOptions,
+    ValidatorConstraint,
+    ValidatorConstraintInterface,
+    ValidationArguments,
+    isString,
+    maxLength,
+    isNumber,
+    min,
+    max,
+    isBoolean
+} from 'class-validator';
+import { constants } from '../../common/constant';
+import { NodeNotifyObject, NodePermissions } from '../../consumer/rpc/v3_3';
+
+function validateTriggerN(triggerN: string | number | boolean) {
+    return (isNumber(triggerN) || isString(triggerN) || isNumber(triggerN));
+}
+
+function validateTriggerT(triggerT: number) {
+    return (isNumber(triggerT) && min(triggerT, 0) && max(triggerT, 4294967295));
+}
+
+function validateStringObject(value: string) {
+    return (isString(value) &&
+    maxLength(value, constants.DT_STRING_MAXLEN));
+}
+
+function validatePermissionsObject(permissions: NodePermissions) {
+    return (isBoolean(permissions.config) ||
+        isBoolean(permissions.read) ||
```

Comment:
should it be && instead of || ? Or its enoughthat one of them is boolean?

Reply:
We expect at least one of those permissions (config, read, write) to be true. It is used for changing node permissions. If all of those are false, it shall not return a NodeNotify object containing permissions.